packages/api-client/src/hooks/useResponseBody.ts

```diff
@@ -1,37 +1,44 @@
 import { extractFilename } from '@/libs/extractAttachmentFilename'
-import { computed } from 'vue'
+import { computed, isRef, ref } from 'vue'
 import MimeType from 'whatwg-mimetype'
+import type { Ref } from 'vue'
 
 /**
  * Processes the response body of an HTTP request.
  * Extracts MIME type, attachment filename, and generates a data URL.
  */
 export function useResponseBody(props: {
-  data: unknown
-  headers: { name: string; value: string; required: boolean }[]
+  data: Ref<unknown> | unknown
+  headers:
+    | Ref<{ name: string; value: string; required: boolean }[]>
+    | { name: string; value: string; required: boolean }[]
```

Comment:
```suggestion
  data: MaybeRef<unknown> 
  headers: MaybeRef<{ name: string; value: string; required: boolean }[]>
```

got some utility types for this

Reply:
on second thought I would just force them to be refs, in which case do we ever not want refs?